### ISSUE TITLE
[QUESTION] is GCHP v 13.0.2 installation  with spack working?

### ISSUE THREAD

**Issue by mtsivlidou** [geoschem/GCHP#113]
Hello, 
I managed to install GCHP v 13.0.2 via spack, with the help of the support of the cluster. 
We use intelmpi/2019.4.243 and the rest of the dependencies are built through spack. 

In the log file I see the info related to runConfig sourcing but the run breaks immediately after sourcing, without any error in the slurm or in the log file. Do you have any idea why this might happen?
Thank you in advance, 
Maria

**Comment by WilliamDowns** [comment]
Hi Maria, sorry to hear this is causing trouble. 13.0.2 via Spack definitely works on other setups. 

Can you post the slurm and log files anyway? Could you also post the output of `spack load gchp@13.0.2; spack find --loaded`? Also, did you copy the `gchp` executable from the Spack-installed GCHP directory?

**Comment by mtsivlidou** [comment]
Thank you for your answer. 
Please find attached the log and the slurm file: 
[slurm-12039887.out.txt](https://github.com/geoschem/GCHP/files/6771478/slurm-12039887.out.txt)
[gchp.log.txt](https://github.com/geoschem/GCHP/files/6771479/gchp.log.txt)

Yes, I copied the executable from the GCHP directory. I tried it also to make a symbolic link but it doesnt work. 

I do not load gchp though spack but with module:
you can find the explanation from the support here: 
[expl.txt](https://github.com/geoschem/GCHP/files/6771534/expl.txt)

Maria

**Comment by WilliamDowns** [comment]
Hmmm, I don't know anything about how that module process would affect loading proper libraries. I don't see anything definitely wrong in that `ldd` output, though I haven't been able to compile successfully in the past when using `mkl` instead of `openblas` libraries when installing through Spack.

Can you also post your Slurm run script, as well as any environment file you load through the Slurm script?

You can also try putting `unset I_MPI_PMI_LIBRARY` in your environment file / slurm script and trying to run again - this was a recommended step when running GCHP using `mpirun` for Intel MPI versions around the version you're using.

**Comment by mtsivlidou** [comment]
Hello, 
Please find attached the env and the slurm script:
[simple_job.sh.txt](https://github.com/geoschem/GCHP/files/6775978/simple_job.sh.txt)
[gchpspack.env.txt](https://github.com/geoschem/GCHP/files/6775980/gchpspack.env.txt)

I included unset I_MPI_PMI_LIBRARY but it dont see any difference. 
Maria

**Comment by WilliamDowns** [comment]
You would need to add the `unset I_MPI_PMI_LIBRARY` line below the `export I_MPI_PMI_LIBRARY=/usr/local/lib64/libpmi2.so` line in your job script (or just comment out the `export I_MPI_PMI_LIBRARY` line). 

After trying that, is it possible for you to do a local test run without using slurm? You can try this by sourcing your environment file and doing `mpirun -np 6 ./gchp`. You'll want to set the run to c24 resolution for this in `runConfig.sh` and change the number of cores / number of cores per node to 6 in that file as well.

Also, can you upload `/scratch/cnt0022/lat0005/tsivlidou/env_spack.0.16.2`?

**Comment by mtsivlidou** [comment]
Hello, 
Sorry I havent sent you the file. We were trying different compilers to see if there is any improvement. With openmpi via spack there is some progress. Now it reads the emission input files and it breaks before starting the vertical interpolation of the emissions. 

srun: debug2: eio_message_socket_accept: got message connection from 172.30.12.58:47668 12
srun: debug2: received task exit
srun: Received task exit notification for 9 tasks of step 12063041.0 (status=0x0006).
srun: error: n3163: tasks 0-1,6-7,10,13,15-17: Aborted
srun: debug:  task 0 done

Could this error be related to the compilation or the system? 

Thank you 
Maria
[cines_test_with_openmpi.slurm.txt](https://github.com/geoschem/GCHP/files/6823232/cines_test_with_openmpi.slurm.txt)

[gchp-13.0.2-cines-test-openmpi-4.1.1-12063041.log.txt](https://github.com/geoschem/GCHP/files/6823233/gchp-13.0.2-cines-test-openmpi-4.1.1-12063041.log.txt)

[HEMCO.log.txt](https://github.com/geoschem/GCHP/files/6823236/HEMCO.log.txt)

[slurm.out.txt](https://github.com/geoschem/GCHP/files/6823239/slurm.out.txt)

**Comment by lizziel** [comment]
Hi Maria, I think the underlying issue is indicated here in the slurm output log:

> [n3165:8070 :0]  ud_channel.c:772  Fatal: UD timeout sending to n3164:63789 (after 600.00 seconds)

According to the GCHP log this is encountered during the "prefetch" step in MAPL ExtData within GCHP. I believe this is an issue with MPI but I have not seen it before. I am pinging @tclune and @bena-nasa (developers of MAPL) to see if they have seen this.

**Comment by tclune** [comment]
Adding @mathomp4 as he's our local guru on low-level MPI oddities.

**Comment by tclune** [comment]
The error does not look familiar to me.    Quite possibly it is simply that you are putting too much stress on the MPI layer with your large input set.   If so, hopefully there are MPI variables that can be set to buffer things better;  failing that we could probably add some option in ExtData and pFIO to limit activity in any given burst.

**Comment by lizziel** [comment]
Maria, could you try a transport tracer simulation (create new run directory) and see if this gets past the issue? This would greatly reduce the number of inputs and test the theory that the issue is a large input set. One node and C24 would be sufficient for the test, although if it works you should then try your current 6 node and C48 configuration.

**Comment by mtsivlidou** [comment]
Okay I will do this. It has to be a monthly simulation right? I will put 1 node and 24 cores.

**Comment by mathomp4** [comment]
I don't recognize that error either. I see something like it as a Known Error in HPC-X with OpenSHMEM, but that's way outside this run. 

And I'm not sure Open MPI has any good ways to adjust timeouts...

One thing I will say is we usually run GEOS with unlimited stacksize. In your logs I see:
```
ulimit -s 10240              # stacksize
```
but then again, this error doesn't look like a stack error. This sort of error:
```
mxm_handle_error
```
is the sort of thing that makes me go look at how Open MPI was built. I know the MXM was deprecated in Open MPI in favor of UCX but I'm not sure how it was built.

It does seem to have crashed in the regridder:
```
25 0x00000000020ca8d0 c_esmc_regrid_create_()  ??:0
26 0x0000000001f8f6ed __esmf_regridmod_MOD_esmf_regridstore()  ??:0
27 0x0000000001d70c93 __esmf_fieldregridmod_MOD_esmf_fieldregridstorenx()  ??:0
srun: debug2: Leaving  _file_write
srun: debug2: Called _file_readable
srun: debug2: Called _file_writable
srun: debug2: Called _file_writable
srun: debug2: Entering _file_write
28 0x00000000013d515c __mapl_esmfregriddermod_MOD_create_route_handle()  /tmp/tsivlidou/spack-stage-gchp-13.0.2-hgyhsblb7tubc64ovb5uloaxyxif3uvb/spack-s
rc/src/MAPL/base/MAPL_EsmfRegridder.F90:1451
29 0x00000000013d8799 __mapl_esmfregriddermod_MOD_initialize_subclass()  /tmp/tsivlidou/spack-stage-gchp-13.0.2-hgyhsblb7tubc64ovb5uloaxyxif3uvb/spack-s
rc/src/MAPL/base/MAPL_EsmfRegridder.F90:1310
30 0x00000000012346d8 __mapl_abstractregriddermod_MOD_initialize_base()  /tmp/tsivlidou/spack-stage-gchp-13.0.2-hgyhsblb7tubc64ovb5uloaxyxif3uvb/spack-s
rc/src/MAPL/base/MAPL_AbstractRegridder.F90:970
```
but that could be something as much as "out of memory"?

**Comment by mtsivlidou** [comment]
I started an 1-hour transport tracer simulation and the past 5 minutes it is stuck here without breaking:

 ExtData Run_: Start
 ExtData Run_: READ_LOOP: Start
 ExtData Run_: ---PopulateBundle
 ExtData Run_: ---prefetch

**Comment by lizziel** [comment]
Has it gotten any further? It does not need to be a month-long simulation. The regridding of the input files is independent of simulation duration.

**Comment by mtsivlidou** [comment]
It was one hour simulation. It took 5 min more less to reach the point I mentioned, and then it was stuck for 20 -25min without moving further so I canceled the job.

**Comment by LiamBindle** [comment]
@mtsivlidou I just noticed this. Just so you know hanging here

```
ExtData Run_: Start
ExtData Run_: READ_LOOP: Start
ExtData Run_: ---PopulateBundle
ExtData Run_: ---prefetch
```
for quite a while is "expected". This is where ExtData is finding all the files on disk, and it does take quite a while. In my experience it seems to get worse the more nodes I use. On a few nodes it might take ~20 minutes some times, but on many nodes (e.g., 20) it sometimes takes several hours. 

This is off topic so I opened https://github.com/geoschem/GCHP/issues/132
